From 54987d445ea714b467d901b7daf8c09ed0644189 Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@freedesktop.org>
Date: Thu, 12 Mar 2020 14:07:23 +0100
Subject: [PATCH] build: Adapt to backwards incompatible change in GNU Make 4.3

GNU Make 4.3 has a backwards incompatible change affecting the use of
number signs or hashes (ie., #) inside function invocations. See:
https://lists.gnu.org/archive/html/info-gnu/2020-01/msg00004.html

In this case, it would expand the '\#' in the '\n\#include \"$(h)\"'
argument to the foreach call to '\#', not '#'. This would lead to
spurious backslashes in front of the '#include' directives in the
generated fs-enumtypes.c file.

Spotted by Ernestas Kulik.

https://gitlab.freedesktop.org/farstream/farstream/-/merge_requests/4
---
 common-modified/gst-glib-gen.mak | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/common-modified/gst-glib-gen.mak b/common-modified/gst-glib-gen.mak
index 14f1ba37..2247a41c 100644
--- a/common-modified/gst-glib-gen.mak
+++ b/common-modified/gst-glib-gen.mak
@@ -6,7 +6,8 @@
 #glib_gen_prefix=gst_color_balance
 #glib_gen_basename=colorbalance
 
-enum_headers=$(foreach h,$(glib_enum_headers),\n\#include \"$(h)\")
+hash:=\#
+enum_headers=$(foreach h,$(glib_enum_headers),\n$(hash)include \"$(h)\")
 
 # these are all the rules generating the relevant files
 $(glib_gen_basename)-enumtypes.h: $(glib_enum_headers)
-- 
GitLab

